example/ipss_app/parse_and_plot.py

```diff
@@ -0,0 +1,208 @@
+
+parse_and_plot.py is a simple script to validate the performance of the 
+ipss cell. No input validation is performed.
+
+- Parse gdf spike file
+- Performs a binning operation for the first 1000 ms 
+  - with 1 ms binning 
+  - averages the found rates assuming a total of 10000 cells.
+- Displays a figure with the histogram of the instantatious spike rate for
+the whole group, including a target rate.
+
+Default target rate: 
+ hz.|                     
+ 240|     _-_             
+    |    -   -  -         
+    |   -     -- -        
+ 0  |__-__________-__     
+      100        900   ms 
+
+Todo:
+ - Add statistical test for correct output for automated validation
+"""
+
+default_time_rate_pairs = [[0.0   , 0.0 * multiplier],
+                           [50.0  , 0.0 * multiplier],
+                           [100.0 , 1.0 * multiplier],
+                           [200.0 , 5.0 * multiplier],
+                           [300.0 , 7.0 * multiplier],
+                           [400.0 , 8.0 * multiplier],
+                           [500.0 , 7.0 * multiplier],
+                           [600.0 , 3.0 * multiplier],
+                           [700.0 , 3.0 * multiplier],
+                           [750.0 , 5.0 * multiplier],
+                           [800.0 , 2.5 * multiplier],
+                           [900.0 , 0.0 * multiplier],
+                           [1000.0, 0.0 * multiplier]]
+
+def parse_spike_file(path):
+    """ 
+    Simple spike file parsing function. Assumes no errors and might fail 
+    silently on errors
+    """
+    spikes_per_cell = collections.defaultdict(list)
+    with open(path , "r") as f:
+        for line in f:
+            tokens = line.split(",")
+            try:
+                gid = int(tokens[0].strip())
+                time = float(tokens[1].strip())
+            except:
+                print "Failed parsing:", tokens[0], ", ", tokens[1]               
+                exit(2)
+
+            spikes_per_cell[gid].append(time)
+
+    return spikes_per_cell
+
+def parse_time_rate(path):
+    """ 
+    Simple spike file parsing function. Assumes no errors and might fail 
+    silently on errors
+    """
+
+    time_rate = []
+
+    with open(path , "r") as f:
+        for line in f:
+            tokens = line.split(",")
+            try:
+                time = float(tokens[0].strip())
+                rate = float(tokens[1].strip())
+            except:
+                print "Failed parsing:", tokens[0], ", ", tokens[1]               
+                exit(2)
+
+            time_rate.append([time, rate])
+
+    return time_rate
+
+def binning(spikes_per_cell, bin_size = 1.0, duration = 1000.0):
+    """
+    Perform a simple binning on the collected spikes before duration
+    Assuming the times are in ms resolution
+    bin_size (ms)
+    duration (ms)
+    """
+    n_bins = int(math.floor(duration / bin_size))
+    bins=[0]*n_bins
+
+    for gid, spikes in spikes_per_cell.items():
```

Comment:
```
for spike in (s for _, spikes in spikes_per_cell.items() for s in spikes):
  ...
```

Maybe not... but maybe just replace `gid` with `_` since it's not used.